netjson-monitoring.lua

```diff
@@ -236,45 +252,108 @@ function is_excluded(name)
   end
 end
 
+function find_default_gateway(routes)
+  for i = 1, #routes do
+    if routes[i].target == '0.0.0.0' then
+      return routes[i].nexthop
+    end
+  end
+  return nil
+end
+
+-- collect device data
+network_status = ubus:call('network.device', 'status', {})
+wireless_status = ubus:call('network.wireless', 'status', {})
+interface_data = ubus:call('network.interface', 'dump', {})
+nixio_data = nixio.getifaddrs()
+
+
+function new_address_array(address, interface, family)
+    proto = interface['proto']
+    if proto == 'dhcpv6' then
+        proto = 'dhcp'
+    end
+    new_address = {
+        address = address['address'],
+        mask = address['mask'],
+        proto = proto,
+        family = family,
+        gateway = find_default_gateway(interface.route),
+    }
+    if next(interface['dns-search']) then
+        new_address.dns_search = interface['dns-search']
+    end
+    if next(interface['dns-server']) then
+        new_address.dns_server = interface['dns-server']
```

Comment:
DNS info should be added to the main netjson object, not the interface object

Reply:
@nemesisdesign It should be in the interface object rather than in the adresses object, right? Otherwise I don't understand it

Reply:
> @nemesisdesign It should be in the interface object rather than in the adresses object, right? I changed it to this way. Otherwise I don't understand it

Not on the interface but in the main `netjson` table:

```lua
netjson.dns_search = dns_info.search
netjson.dns_servers = dns_info.servers  
```

Should be `servers` rather than `server`.